client/state/sites/domains/assembler.js

```diff
@@ -9,6 +9,7 @@ export const createSiteDomainObject = domain => {
 		autoRenewing: Boolean( domain.auto_renewing ),
 		blogId: Number( domain.blog_id ),
 		canSetAsPrimary: Boolean( domain.can_set_as_primary ),
+		canUseDesignatedAgent: Boolean( domain.can_use_designated_agent ),
```

Comment:
You need to add this field to `lib/domains/assembler.js` as well :) Otherwise, it's just not working (the WHOIS edit form uses that assembler, not the Redux one).
(I know, it's confusing and crappy dev experience)